goaviatrix/client_proxy_config.go

```diff
@@ -0,0 +1,92 @@
+package goaviatrix
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"strings"
+)
+
+type ClientProxyConfig struct {
+	HttpProxy          string `form:"http_proxy,omitempty" json:"http_proxy,omitempty"`
+	HttpsProxy         string `form:"https_proxy,omitempty" json:"https_proxy,omitempty"`
+	ProxyCaCertificate string
+}
+
+type ClientProxyConfigResp struct {
+	Return  bool
+	Results ClientProxyConfig
+	Reason  string
+}
+
+func (c *Client) CreateClientProxyConfig(clientProxyConfig *ClientProxyConfig) error {
+	action := "apply_proxy_config"
+	if clientProxyConfig.ProxyCaCertificate != "" {
+		params := map[string]string{
+			"CID":         c.CID,
+			"action":      action,
+			"http_proxy":  clientProxyConfig.HttpProxy,
+			"https_proxy": clientProxyConfig.HttpsProxy,
+		}
+
+		files := []File{
+			{
+				Path:      clientProxyConfig.ProxyCaCertificate,
+				ParamName: "server_ca_cert",
+			},
+		}
```

Comment:
This will not work with file content, this is for reading a file path.
To use file content you would use the UseFileContent option. See Shuo's filebeat_forwarder resource as an example.
```

	var files []File

	if r.TrustedCAFile != "" {
		ca := File{
			ParamName:      "trusted_ca",
			UseFileContent: true,
			FileName:       "ca.pem", // fake name for ca
			FileContent:    r.TrustedCAFile,
		}
		files = append(files, ca)
	}
```